Binary files orig/XML/src/.XMLEventParse.c.swp and patched/XML/src/.XMLEventParse.c.swp differ
diff -Nru orig/XML/src/DocParse.c patched/XML/src/DocParse.c
--- orig/XML/src/DocParse.c	2023-11-29 11:40:43.000000000 +0100
+++ patched/XML/src/DocParse.c	2024-01-18 12:47:34.189541100 +0100
@@ -46,7 +46,13 @@
 USER_OBJECT_
 RS_XML(libxmlVersionRuntime)(void)
 {
-    return(mkString(*__xmlParserVersion()));
+    return(mkString(
+#if LIBXML_VERSION < 21200
+		*__xmlParserVersion()
+#else
+		xmlParserVersion
+#endif
+	));
 }
 
 
diff -Nru orig/XML/src/XMLEventParse.c patched/XML/src/XMLEventParse.c
--- orig/XML/src/XMLEventParse.c	2022-10-02 13:12:27.000000000 +0200
+++ patched/XML/src/XMLEventParse.c	2024-01-18 13:00:57.329883800 +0100
@@ -12,6 +12,12 @@
 #include <libxml/parserInternals.h>
 #endif
 
+#ifdef _WIN32
+/* on Unix, there is a configure check */
+# if LIBXML_VERSION < 20627 || LIBXML_VERSION > 21100
+#  define NO_CHECKED_ENTITY_FIELD 
+# endif
+#endif
 
 static USER_OBJECT_ createSAX2AttributesList(const xmlChar **attributes, int nb_attributes, int nb_defaulted, const xmlChar *encoding);
 
